php-checks/src/main/java/org/sonar/php/checks/HardCodedCredentialsCheck.java

```diff
@@ -74,6 +99,27 @@
       .collect(Collectors.toList());
   }
 
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String functionName = tree.callee().toString();
+    if (CONNECT_FUNCTIONS.containsKey(functionName)) {
+      checkArgument(tree, CONNECT_FUNCTIONS.get(functionName));
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private void checkArgument(FunctionCallTree tree, int argNumber) {
+    if (argNumber > tree.arguments().size()) {
+      return;
+    }
+
+    ExpressionTree arg = tree.arguments().get(argNumber - 1);
+
+    if(arg.is(Kind.REGULAR_STRING_LITERAL) && ((LiteralTree)arg).value().length() - 2 >= MIN_LENGTH_OF_HARDCODED_PASSWORD) {
```

Comment:
I don't understand the condition on the length.

Reply:
Removed it. I wasn't sure if it makes sense to check if the string literal is empty. But yes, an empty string literal is still a hardcoded empty password in these locations :)

Reply:
Sorry, I was not clear. Your code seemed to exclude cases where the length is less than 4 (because `MIN_LENGTH_OF_HARDCODED_PASSWORD` was set to 2).

I think that this rule should ignore empty passwords because there's another rule for that: https://rules.sonarsource.com/php/RSPEC-2115. That makes me think that we could have tried to share some code between the 2 rules.

Reply:
Added the check for emptiness by using the method isEmptyStringLiteral() that already existed in that class. My initial thought was to check if the string contains 2 characters or more  (excluding the quotes) and then I noticed that ((LiteralTree)arg).value() still contains the quotes, hence the "-2" :) 

Indeed, S2115 is very similar.